src/System.Text.Json/tests/Serialization/Array.ReadTests.cs

```diff
@@ -78,6 +78,15 @@ public static void ReadByteArrayFail()
             Assert.Throws<JsonException>(() => JsonSerializer.Parse<byte[]>(@"""A==="""));
         }
 
+        [Theory]
+        [InlineData(typeof(int[]), @"[{}]")]
+        [InlineData(typeof(Dictionary<string, int[]>), @"{""test"": [{}]}")]
+        public static void InvalidJsonForArrayShouldFail(Type type, string json)
+        {
+            // These fail because the int converter sees the StartObject token.
+            Assert.Throws<JsonException>(() => JsonSerializer.Parse(json, type));
+        }
```

Comment:
can you also remove these https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/tests/Serialization/DictionaryTests.cs#L220 &
https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/tests/Serialization/DictionaryTests.cs#L228